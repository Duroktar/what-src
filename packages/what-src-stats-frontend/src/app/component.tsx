import * as React from 'react'
import { withTheme } from 'emotion-theming'
import { Text } from 'rebass'
import { When } from 'react-if'
import Head from './components/head'
import Nav from './components/nav'
import { Arrow } from './components/arrow'
import { Info } from './components/info'
import { Loading } from './components/loading'
import { Logo } from './components/logo'
import { Toast } from './components/toast'
import { BackGroundComponent } from './components/backgrounds'
import { useComponentState } from './hooks'
import { IMG_CONFIG_BASIC, IMG_BASE_URL } from '../constants'

type AppProps = {
  setTheme: (theme: string) => void
  currentTheme: string
  locale: string
}

export const App =
React.memo(withTheme((props: AppProps) => {
  const { t, error, ...state } = useComponentState()
  return (
    <BackGroundComponent background={state.background}>
      <div className="app fullscreen">
        <Head title="Home" />
        <Nav />
        <div className="msg container">
          <h1 className="row logo">
            <Logo />
          </h1>
          <When condition={!!state.durationText}>
            <span className="row count">
              <Text fontSize={27} color='primary'>
                {t('Has saved approximately')}
              </Text>
              <Text fontSize={27} color='primary'>
                <b>{state.humanTime}</b>
              </Text>
              <Text fontSize={27} color='primary'>
                {t('of time.')}
              </Text>
              <Text fontSize={27} color='primary'>
                  ({t('About')} <b>{state.countUp} {t('seconds')}</b>..)
              </Text>
              <br />
              <Text fontSize={27} color='primary' sx={state.niceStyle}>
                <span><i>{t('Nice.')}</i></span>
              </Text>
            </span>
          </When>
          <span className="row status">
            <When condition={!error}>
              <span className="error">{error}</span>
            </When>
            <When condition={state.loading}>
              <span className="loading">
                <Loading />
              </span>
            </When>
          </span>
        </div>
        <div className="example">
          <div id='arrow'><Arrow down={!state.toast} /></div>
          <Toast
            id="toast"
            onClick={state.toggleToast}
            src={IMG_BASE_URL + IMG_CONFIG_BASIC}
          />
        </div>
        <div className='footer'>
          <div
            id="refresh-btn"
            role="button"
            tabIndex={0}
            onClick={state.handleRefresh}
            className={!error ? 'hidden' : undefined}
          >
            {t('Refresh')}
          </div>
          <button
            id="info-bubble"
            role="button"
            name="information"
            tabIndex={-1}
            onClick={state.handleRefresh}
          >
            <Info txt={t('INFO-TXT')} />
          </button>
        </div>
      </div>
    </BackGroundComponent>
  )
}))
