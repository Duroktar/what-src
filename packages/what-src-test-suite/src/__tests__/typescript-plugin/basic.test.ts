
import * as ts from 'typescript'
import { createTransformer, WhatSrcTsTransformer } from '@what-src/typescript-plugin'
import { basicComponents } from '../../fixtures/component.fixture'

const compilerOptions: ts.CompilerOptions = {
  module: ts.ModuleKind.CommonJS,
  jsx: ts.JsxEmit.React,
  target: ts.ScriptTarget.ESNext,
  importHelpers: true,
}

const transformerOptions = {
  cacheLocOverride: '/CACHE_LOCATION/OVERRIDE',
  createCacheDir: false,
  createCacheFile: false,
}

describe('ts-transformer plugin basic tests', () => {
  it('works as advertised', () => {
    const result = ts.transpileModule(basicComponents, {
      compilerOptions,
      transformers: { before: [createTransformer(transformerOptions)] },
    })

    expect(result.outputText).toMatchSnapshot()
  })

  it('can optionally take a ts.Loader argument', () => {
    expect(createTransformer({}, transformerOptions)).not.toBeNull()
  })

  it('can also be disabled if needed', () => {
    const Transformer = WhatSrcTsTransformer as any
    const logger = { log: () => null, error: () => null }
    const old = process.env.NODE_ENV

    process.env.NODE_ENV = 'production'

    const plug = new Transformer({}, null, null, null, logger)
    expect(plug.disabled).toEqual(true)

    process.env.NODE_ENV = old

    const plug2 = new Transformer({}, null, null, null, logger)
    expect(plug2.disabled).toEqual(false)
  })
})
