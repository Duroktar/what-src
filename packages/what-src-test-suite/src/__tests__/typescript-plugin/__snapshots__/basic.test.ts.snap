// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-transformer plugin basic tests works as advertised 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
require(\\"@what-src/runtime-cache\\").default;
const React = require(\\"react\\");
class TestClassComponent extends React.Component {
    render() {
        return (React.createElement(\\"div\\", { \\"data-what-src\\": '1' },
            React.createElement(\\"p\\", { \\"data-what-src\\": '2' }, \\"some p thing\\")));
    }
}
const TestFunctionalComponent = () => (React.createElement(\\"div\\", { \\"data-what-src\\": '3' },
    React.createElement(\\"p\\", { \\"data-what-src\\": '4' }, \\"some p thing\\")));
const TestMemoizedComponent = React.memo(() => (React.createElement(\\"div\\", { \\"data-what-src\\": '5' },
    React.createElement(\\"p\\", { \\"data-what-src\\": '6' }, \\"some p thing\\"))));
"
`;
